File: AllSpice/Services/IngredientsService.cs
namespace AllSpice.Services
{
    public class IngredientsService
    {
        private readonly IngredientsRepository _repo;

        public IngredientsService(IngredientsRepository repo)
        {
            _repo = repo;
        }

        internal Ingredient CreateIngredient(Ingredient ingredientData)
        {
            Ingredient ingredient = _repo.Insert(ingredientData);
            return ingredient;
        }

        internal List<Ingredient> GetRecipeIngredients(int recipeId)
        {
            List<Ingredient> ingredients = _repo.GetRecipeIngredients(recipeId);
            return ingredients;
        }
    }
}

File: AllSpice.client/src/services/IngredientsService.js
import { AppState } from "../AppState"
import { Ingredient } from "../models/Ingredient"
import { logger } from "../utils/Logger"
import { api } from "./AxiosService"


class IngredientsService {

    async createIngredient(ingredientData) {
        const res = await api.post('api/ingredients', ingredientData)
        logger.log('[Created ingredient]', res.data)
        AppState.ingredients.push(new Ingredient(res.data))
        return res.data
    }

    async getIngredientsByRecipeId(recipeId) {
        const res = await api.get(`api/recipes/${recipeId}/ingredients`)
        logger.log('[Recipe Ingredients]', res.data)
        AppState.ingredients = res.data
    }

    async deleteIngredient(ingredientId) {
        const res = await api.delete(`api/ingredients/${ingredientId}`)
        logger.log('[Delete Ingredient]', res.data)
        AppState.ingredients = AppState.ingredients.filter(i => i.ingredientId != ingredientId)
        logger.log(AppState.ingredients)
    }
}

export const ingredientsService = new IngredientsService()